### PR TITLE
Dedicated guides on cloud credentials

### PR DESCRIPTION
**Pull request by jmcx**
Docs for
* [Add support for GKE workload identity](https://github.com/triggermesh/triggermesh/pull/1223)
* [GCP Cloud Services - Custom KSA Names for GKE](https://github.com/triggermesh/triggermesh/pull/1372)
* [AWS sources and targets cross-account/remote IAM support](https://github.com/triggermesh/triggermesh/pull/1400)
* Annotation for AWS components with reconciliation logic to use the IAM role
* ...

### CODE REVIEW

**Review by tzununbekov — CHANGES_REQUESTED** [review]
I haven't tested the configuration steps or the manifests in this update, but I assume that they are OK. Except for a couple of small things, this is looking good!

File: docs/guides/credentials/gcpcredentials.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  annotations:
+    iam.gke.io/gcp-service-account: my-sa@my-project.iam.gserviceaccount.com
+spec:
+  topic: projects/my-project/topics/my-topic
+  kubernetesServiceAccount: <my-custom-ksa-name>
+  sink:
+    ref:
+      apiVersion: eventing.triggermesh.io/v1alpha1
+      kind: RedisBroker
+      name: triggermesh
```
[the remark this answers is not in the document]
Comment (jmcx) [reply docs/guides/credentials/gcpcredentials.md:84]: Thanks, update to the below. Assuming we don't need to point to the GSA anymore, just the KSA in this case?

```yaml
apiVersion: sources.triggermesh.io/v1alpha1
kind: GoogleCloudPubSubSource
metadata:
  name: sample
spec:
  topic: projects/my-project/topics/my-topic
  auth:
    kubernetesServiceAccount: <my-custom-ksa-name>
  sink:
    ref:
      apiVersion: eventing.triggermesh.io/v1alpha1
      kind: RedisBroker
      name: triggermesh
```

[the remark this answers is not in the document]
Comment (tzununbekov) [reply docs/guides/credentials/gcpcredentials.md]: We still need GSA.

[the remark this answers is not in the document]
Comment (jmcx) [reply docs/guides/credentials/gcpcredentials.md]: thx, fixed

**Review by tzununbekov — APPROVED** [review]

**Pull request merged**